central/networkgraph/entity/datastore/datastore_impl_test.go

```diff
@@ -633,3 +640,1023 @@ func (suite *NetworkEntityDataStoreTestSuite) expectPushExternalNetworkEntitiesT
+		},
+		"Cluster partial access cannot get global entity": {
+			contextName:    sacTestUtils.Cluster1NamespaceAReadWriteCtx,
+			entityID:       entity1ID.String(),
+			expectedExists: false,
+		},
+		"Cluster partial access cannot get cluster entity": {
+			contextName:    sacTestUtils.Cluster1NamespaceAReadWriteCtx,
+			entityID:       entity2ID.String(),
+			expectedExists: false,
+		},
+		"Cluster partial access cannot get missing entity": {
+			contextName:    sacTestUtils.Cluster1NamespaceAReadWriteCtx,
+			entityID:       entity3ID.String(),
+			expectedExists: false,
+		},
+		"Other cluster access cannot get global entity": {
+			contextName:    sacTestUtils.Cluster2ReadWriteCtx,
+			entityID:       entity1ID.String(),
+			expectedExists: false,
+		},
+		"Other cluster access cannot get cluster entity": {
+			contextName:    sacTestUtils.Cluster2ReadWriteCtx,
+			entityID:       entity2ID.String(),
+			expectedExists: false,
+		},
+		"Other cluster access cannot get missing entity": {
+			contextName:    sacTestUtils.Cluster2ReadWriteCtx,
+			entityID:       entity3ID.String(),
+			expectedExists: false,
+		},
+	} {
+		s.Run(name, func() {
+			ctx := testContexts[tc.contextName]
+			exists, testErr := s.ds.Exists(ctx, tc.entityID)
+			s.NoError(testErr)
+			s.Equal(tc.expectedExists, exists)
+		})
+	}
+}
+
+func (s *NetworkEntityDataStoreTestSuite) TestGetIDsSAC() {
+	entityIDs, _ := s.setupSACReadSingleTest()
+	entity1ID := entityIDs[0]
+	entity2ID := entityIDs[1]
+	testContexts := sacTestUtils.GetNamespaceScopedTestContexts(context.Background(), s.T(), resources.NetworkGraph)
+
+	allIDs := []string{entity1ID.String(), entity2ID.String()}
+	clusterIDs := []string{entity2ID.String()}
+	noIDs := []string{}
+
+	for name, tc := range map[string]struct {
+		contextName string
+		expectedIDs []string
+	}{
+		"All access can get all entity IDs": {
+			contextName: sacTestUtils.UnrestrictedReadWriteCtx,
+			expectedIDs: allIDs,
+		},
+		"Full read access can get all entity IDs": {
+			contextName: sacTestUtils.UnrestrictedReadCtx,
+			expectedIDs: allIDs,
+		},
+		"Cluster full access can only get cluster entity IDs": {
+			contextName: sacTestUtils.Cluster1ReadWriteCtx,
+			expectedIDs: clusterIDs,
+		},
+		"Cluster partial access cannot get any entity ID": {
+			contextName: sacTestUtils.Cluster1NamespaceAReadWriteCtx,
+			expectedIDs: noIDs,
+		},
+		"Other cluster access cannot get any entity ID": {
+			contextName: sacTestUtils.Cluster2ReadWriteCtx,
+			expectedIDs: noIDs,
+		},
+	} {
+		s.Run(name, func() {
+			ctx := testContexts[tc.contextName]
+			fetchedIDs, testErr := s.ds.GetIDs(ctx)
+			s.NoError(testErr)
+			s.ElementsMatch(tc.expectedIDs, fetchedIDs)
+		})
+	}
+}
+
+func (s *NetworkEntityDataStoreTestSuite) TestGetEntitySAC() {
+	entityIDs, entities := s.setupSACReadSingleTest()
+	entity1ID := entityIDs[0]
+	entity2ID := entityIDs[1]
+	entity3ID := entityIDs[2]
+	globalEntity := entities[0]
+	clusterEntity := entities[1]
```

Comment:
That part is hard for me but I have no idea how ti make it better

Reply:
Resolved with the refactor.